Card.cpp

```diff
@@ -0,0 +1,62 @@
+//member functons for the Card class defined here
+#include <iostream>
+#include <cassert>
+
+#include "Card.h"
+
+int Card::getCardValue() const
+{
+    switch (m_rank)
+    {
+        case Card::Rank::rank_2:        return 2;
+        case Card::Rank::rank_3:        return 3;
+        case Card::Rank::rank_4:        return 4;
+        case Card::Rank::rank_5:        return 5;
+        case Card::Rank::rank_6:        return 6;
+        case Card::Rank::rank_7:        return 7;
+        case Card::Rank::rank_8:        return 8;
+        case Card::Rank::rank_9:        return 9;
+        case Card::Rank::rank_10:       return 10;
+        case Card::Rank::rank_jack:     return 10;
+        case Card::Rank::rank_queen:    return 10;
+        case Card::Rank::rank_king:     return 10;
+        case Card::Rank::rank_ace:      return 11;
+        default:
+            assert(false && "shouldn't happen - getCardValue");
+            return 0;
+    } 
+}
+
+void Card::printCard() const
```

Comment:
Since this printing is for human consumption, it could print something readable like `King of Spades` rather than `K S`.